File: Samples/blazor-add-in/excel-blazor-add-in/excel-blazor-add-in/Pages/Index.razor.cs
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */

using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;

using System.Diagnostics;
using System.Diagnostics.CodeAnalysis;

namespace BlazorAddIn.Pages
{
    /// <summary>
    /// Starter class to demo how to insert `Hello world!` text.
    /// </summary>
    public partial class Index
    {
        [Inject, AllowNull]
        private IJSRuntime JSRuntime { get; set; } 
        private IJSObjectReference JSModule { get; set; } = default!;

        protected override async Task OnAfterRenderAsync(bool firstRender)
        {
            if (firstRender)
            {
                Debug.WriteLine("Hit OnAfterRenderAsync in Index.razor.cs!");
                Console.WriteLine("Hit OnAfterRenderAsync in Index.razor.cs in Console!");
                JSModule = await JSRuntime.InvokeAsync<IJSObjectReference>("import", "./Pages/Index.razor.js");
            }
        }

        /// <summary>
        /// Basic function to invoke inserting `Hello world!` text.
        /// </summary>
        private async Task HelloButton() =>
            await JSModule.InvokeVoidAsync("helloButton");

        [JSInvokable]
        public static string SayHelloIndex(string name)
        {
            return $"Hello Index, {name} from BlazorFunctionFile!";
        }
    }
}

File: Samples/blazor-add-in/archived/excel-blazor-add-in/excel-blazor-add-in/Pages/Index.razor.js
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */
/**
 * Basic function to show how to insert a value into cell A1 on the selected Excel worksheet.
 */
console.log("Loading Index.razor.js");

export function helloButton() {

    console.log("We are now entering function: helloButton");

    return Excel.run(context => {


        // Insert text 'Hello world!' into cell A1.
        context.workbook.worksheets.getActiveWorksheet().getRange("A1").values = [['Hello world!']];

        // sync the context to run the previous API call, and return.
        return context.sync();
    });
}
